### PR TITLE
Handling nodemapping CR spec

### PR DESCRIPTION
**Pull request by nitishkumar4**
## What is the current behavior?

`nodeMapping` CR spec was not handled.

## What is the new behavior?

Controller reads the `nodeMapping` CR spec and adds the mapped nodes as environment variables in `ibm-spectrum-scale-csi` container of daemon set pods.

## Test Results:

Pod description for container `ibm-spectrum-scale-csi`
```json
Environment:
      worker0.whoami.cp.fyre.ibm.com:  worker0
      worker1.whoami.cp.fyre.ibm.com:  worker1
...
```

CR Spec:
```yaml
  nodeMapping:
    - k8sNode: "worker0.whoami.cp.fyre.ibm.com"
      spectrumscaleNode: "worker0"
    - k8sNode: "worker1.whoami.cp.fyre.ibm.com"
      spectrumscaleNode: "worker1"
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,4 @@ testbin/*
 *.swp
 *.swo
 *~
+.vscode/launch.json
```
Comment (madhuthorat) [inline .gitignore:26]: Why is this line added ?

File: controllers/syncer/csi_node.go [hunk 2]
```diff
@@ -245,7 +256,7 @@ func (s *csiNodeSyncer) getEnvFor(name string) []corev1.EnvVar {
 			},
 			envVarFromField("NODE_ID", "spec.nodeName"),
 			envVarFromField("KUBE_NODE_NAME", "spec.nodeName"),
-		}
+		}...)
```
Comment (madhuthorat) [inline controllers/syncer/csi_node.go:259]: Why is this change needed ?

Comment (nitishkumar4) [reply controllers/syncer/csi_node.go:259]: Here I am concating two slices(list),`return append(EnvVars, []corev1.EnvVar{` since the number of element of second slice is not known as slice has dynamic length,  so 3 dots i.e `...`  are added for the go compiler to understand this.
You may check  this link https://www.golangprograms.com/how-to-concatenate-two-or-more-slices-in-golang.html for reference.

File: .gitignore (hunk 1 above)
Comment (nitishkumar4) [reply .gitignore:26]: For debugging with vscode, we need to add launch.json file in the root directory of the branch, added the file to gitignore, so that launch.json file is not added as new file in the repo. Changes to files in this list are ignored by GIT.

Reference to gitignore file: https://www.atlassian.com/git/tutorials/saving-changes/gitignore

**Review by madhuthorat — APPROVED** [review]
LGTM
